controllers/progressivesync_controller.go

```diff
@@ -349,6 +357,23 @@ func (r *ProgressiveSyncReconciler) setSyncedAtAnnotation(ctx context.Context, a
 		if err := r.Client.Update(ctx, &latest); err != nil {
 			return err
 		}
+
+		stageKeyValue := ps.Name + "/" + stageName
+		appKeyValue := ps.Name + "/" + app.Name
+
+		r.SyncedAtStage[appKeyValue] = stageName
+
+		val, ok := r.SyncedAppsPerStage[stageKeyValue]
+		if !ok {
+			r.SyncedAppsPerStage[stageKeyValue] = 1
+		} else {
+			if val+1 > stageMaxTargets {
+				r.SyncedAppsPerStage[stageKeyValue] = stageMaxTargets
```

Comment:
I'm a bit afraid this could mean we  'hide' that we actually synced more apps than we should - but I have no proof to back this claim, so perhaps something we can accept and keep an eye on?